main.tf

```diff
@@ -1,3 +1,280 @@
+}
+
+# creating the namespace to deploy the helm releases to install the WebSphere Liberty Operator
+resource "kubernetes_namespace" "helm_release_operator_namespace" {
+  count = var.add_ibm_operator_catalog == true ? 1 : 0
+
+  metadata {
+    name = var.operator_helm_release_namespace
+  }
+
+  timeouts {
+    delete = "30m"
+  }
+
+  lifecycle {
+    ignore_changes = [
+      metadata[0].annotations,
+      metadata[0].labels
+    ]
+  }
+}
+
+locals {
+  ibm_operator_catalog_version = "v1.25-20240202.161709-9DAF3E648@sha256:92e28be4af60f68c656f52b2445aafcc052fcd0390479b868c5b0ba2d465a25a"
+  ibm_operator_catalog_path    = "icr.io/cpopen/ibm-operator-catalog"
+}
+
+# if add_ibm_operator_catalog is true going on with adding the IBM Operator Catalog source
+resource "helm_release" "ibm_operator_catalog" {
+  depends_on = [data.ibm_container_cluster_config.cluster_config, kubernetes_namespace.helm_release_operator_namespace[0]]
+  count      = var.add_ibm_operator_catalog == true ? 1 : 0
+
+  name              = "ibm-operator-catalog-helm-release"
+  chart             = "${path.module}/chart/${local.ibm_operator_catalog_chart}"
+  namespace         = var.operator_helm_release_namespace
+  create_namespace  = false
+  timeout           = 300
+  dependency_update = true
+  force_update      = false
+  cleanup_on_fail   = false
+  wait              = true
+
+  disable_openapi_validation = false
+
+  set {
+    name  = "image.path"
+    type  = "string"
+    value = local.ibm_operator_catalog_path
+  }
+  set {
+    name  = "image.version"
+    type  = "string"
+    value = local.ibm_operator_catalog_version
+  }
+}
+
+# waiting for the catalog to be configured and correctly pulled
+resource "time_sleep" "wait_catalog" {
+  depends_on = [helm_release.ibm_operator_catalog[0]]
+  count      = var.add_ibm_operator_catalog == true ? 1 : 0
+
+  create_duration = local.sleep_time_catalog_create
+}
+
+# if ws_liberty_operator_target_namespace != null the operator group must be created
+resource "helm_release" "websphere_liberty_operator_group" {
+  count      = var.ws_liberty_operator_target_namespace != null ? 1 : 0
+  depends_on = [time_sleep.wait_catalog[0]]
+
+  name              = "websphere-liberty-operator-group-helm-release"
+  chart             = "${path.module}/chart/${local.websphere_liberty_operator_group_chart}"
+  namespace         = var.operator_helm_release_namespace
+  create_namespace  = false
+  timeout           = 300
+  dependency_update = true
+  force_update      = false
+  cleanup_on_fail   = false
+  wait              = true
+
+  disable_openapi_validation = false
+
+  set {
+    name  = "operatornamespace"
+    type  = "string"
+    value = var.ws_liberty_operator_namespace
+  }
+
+  set {
+    name  = "operatortargetnamespace"
+    type  = "string"
+    value = var.ws_liberty_operator_target_namespace
+  }
+
+}
+
+resource "kubernetes_namespace" "websphere_liberty_operator_namespace" {
```

Comment:
why not just use `create_namespace = true` in the helm provider

Reply:
the create_namespace refers to the namespace where the helm release is created, while this create the namespace where the operator is deploy, that may be different